Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -476,6 +476,16 @@ class BrowserViewController: UIViewController {
         SearchBarSettingsViewModel.recordLocationTelemetry(for: isBottomSearchBar ? .bottom : .top)
 
         overlayManager.setURLBar(urlBarView: urlBar)
+
+        if !displayedRestoreTabsAlert && crashedLastLaunch() {
```

Comment:
I have a very strong suspicion this can cause problems when the feature flag is off, but maybe moving it could cause even more issues, at least whats there today is a known entity. I will update to add the flag check.

Reply:
With that change all sorts of unit tests started to fail. I could probably dig into them but it scared me I'm breaking something unexpected. Instead I've moved the check inside the new code so it just returns early if there are already tabs in the list.